deploy/setup-external-snapshot-metadata.yaml

```diff
@@ -0,0 +1,150 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: external-snapshot-metadata
```

Comment:
I think `snapshot-metadata-sidecar` would be a more appropriate name, here and as the prefix in the Roles.